htdocs/fourn/class/api_supplier_orders.class.php

```diff
@@ -224,7 +224,7 @@ public function index($sortfield = "t.rowid", $sortorder = 'ASC', $limit = 100,
 	 */
 	public function post($request_data = null)
 	{
-		if (!DolibarrApiAccess::$user->rights->fournisseur->commande->creer || !DolibarrApiAccess::$user->rights->supplier_order->creer) {
+		if (!DolibarrApiAccess::$user->rights->fournisseur->commande->creer) {
```

Comment:
We are preparing the code to split the module "fournisseur" into  2 modules: "supplier_order" and "supplier_invoice". So good test would be:
if ((empty($user->rights->fournisseur->commande->creer) && empty($conf->global->MAIN_USE_NEW_SUPPLIERMOD))
	|| (empty($user->rights->supplier_order->create) && !empty($conf->global->MAIN_USE_NEW_SUPPLIERMOD))) {

We use MAIN_USE_NEW_SUPPLIERMOD to know if we are in old/current mode or in future mode.